src/polyfem/solver/forms/adjoint_forms/VariableToSimulation.cpp

```diff
@@ -24,11 +27,38 @@ namespace polyfem::solver
 			return std::make_unique<DirichletVariableToSimulation>(states, parametrization);
 		else if (type == "pressure")
 			return std::make_unique<PressureVariableToSimulation>(states, parametrization);
+		else if (type == "periodic-shape")
+			return std::make_unique<PeriodicShapeVariableToSimulation>(states, parametrization);
 
 		log_and_throw_adjoint_error("Invalid type of VariableToSimulation!");
 		return std::unique_ptr<VariableToSimulation>();
 	}
 
+	void VariableToSimulation::set_output_indexing(const json &args)
+	{
+		const std::string composite_map_type = args["composite_map_type"];
```

Comment:
wy do you need this?

Reply:
This is moving around the inverse optimization code, the branching was ugly so I moved into each derived class